editoast/src/main.rs

```diff
@@ -185,53 +193,76 @@ async fn generate(
     } else {
         // Retrieve given infras
         for id in args.infra_ids {
-            infras.push(Infra::retrieve(&mut conn, id as i64)?);
+            let infra = match Infra::retrieve(pool.clone(), id as i64).await? {
+                Some(infra) => infra,
+                None => {
+                    return Err(InfraApiError::NotFound {
+                        infra_id: id as i64,
+                    }
+                    .into())
+                }
+            };
+            infras.push(infra);
         }
     };
 
     // Refresh each infras
     for infra in infras {
         println!(
             "🍞 Infra {}[{}] is generating:",
-            infra.name.bold(),
-            infra.id
+            infra.name.clone().unwrap().bold(),
+            infra.id.unwrap()
         );
         let infra_cache = InfraCache::load(&mut conn, &infra)?;
         if infra.refresh(&mut conn, args.force, &infra_cache)? {
-            build_redis_pool_and_invalidate_all_cache(&redis_config.redis_url, infra.id).await;
-            println!("✅ Infra {}[{}] generated!", infra.name.bold(), infra.id);
+            build_redis_pool_and_invalidate_all_cache(&redis_config.redis_url, infra.id.unwrap())
+                .await;
+            println!(
+                "✅ Infra {}[{}] generated!",
+                infra.name.unwrap().bold(),
+                infra.id.unwrap()
+            );
         } else {
             println!(
                 "✅ Infra {}[{}] already generated!",
-                infra.name.bold(),
-                infra.id
+                infra.name.unwrap().bold(),
+                infra.id.unwrap()
             );
         }
     }
     Ok(())
 }
 
-fn import_railjson(
+async fn import_railjson(
     args: ImportRailjsonArgs,
     pg_config: PostgresConfig,
 ) -> Result<(), Box<dyn Error + Send + Sync>> {
     let railjson_file = File::open(args.railjson_path)?;
+    let manager = ConnectionManager::<PgConnection>::new(pg_config.url());
+    let pool = Data::new(Pool::builder().max_size(1).build(manager).unwrap());
     let conn = &mut PgConnection::establish(&pg_config.url()).expect("Error while connecting DB");
```

Comment:
Isn't it weird to use `conn` and `pool` at the same time ?

Reply:
Indeed we should only use the pool here.